### PR TITLE
Rust avr support

### PR DESCRIPTION
**Pull request by Vollbrecht**
examples and templates can be found here: https://github.com/Vollbrecht/wokwi-rust-avr-projects

### CODE REVIEW

File: rust-nostd-avr/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+WORKDIR /home/wokwi
+ENV USER=wokwi
+
+# Check rust version
+RUN cargo --version
+RUN rustup show
+
+# Install extra crates
+# it would be nice if we could get cargo-audit and sccache in some binary form :D this build step take's ages
+RUN cargo install cargo-audit cargo-generate sccache
```
Comment (urish) [inline rust-nostd-avr/Dockerfile:33]: You could build another docker image that precompiled them, publish it on Dockerhub, and then copy it from that image. e.g.

```
COPY --from=user/image /path/to/sccache /path/to/sccache
```

**Comment by urish** [comment]
Thanks! The new builder is up

**Comment by urish** [comment]
The builds are failing, mind taking a look?

https://github.com/wokwi/wokwi-builders/actions/runs/5431351695/jobs/9877663002

**Comment by Vollbrecht** [comment]
currently not at home, can have a look in the evening or tomorrow. 
though looking over it briefly, why is https://github.com/wokwi/wokwi-builders/actions/runs/5431351695/jobs/9877663002#step:3:114 running with `--locked` if it is not specified inside the Dockerfile that way?

**Comment by urish** [comment]
Seems like @SergioGasquez added it here: https://github.com/wokwi/wokwi-builders/commit/2a1967e768ff8c3492f25660a2bb2c18515b399c

**Comment by SergioGasquez** [comment]
>Seems like @SergioGasquez added it here: https://github.com/wokwi/wokwi-builders/commit/2a1967e768ff8c3492f25660a2bb2c18515b399c

The reason was `cargo-generate`, it had an issue where the installation was failing https://github.com/cargo-generate/cargo-generate/issues/942, and using `--locked` solved it, I guess we could remove it since it's now fixed and released, but I've seen this very same error happening in the past more times in `cargo-generate`.

**Comment by Vollbrecht** [comment]
so we try to go without --locked now? I am currently relative busy but i hope i can push all the changes next weekend where i stripped all the building from source, in favor of getting the binary's off all the crates over GH releases

**Comment by SergioGasquez** [comment]
> so we try to go without --locked now? I am currently relative busy but i hope i can push all the changes next weekend where i stripped all the building from source, in favor of getting the binary's off all the crates over GH releases

We can try to go without `--locked`, hopefully the error won't come back in the future. Not sure if you used it as inspiration, but we currently install most of these crates from GH releases in: 
- [templates dockerfile](https://github.com/esp-rs/esp-template/blob/main/.devcontainer/Dockerfile#L32-L44)
- rust-build 
   - [script](https://github.com/esp-rs/rust-build/blob/main/install-rust-toolchain.sh#L475)
   - [Container file](https://github.com/esp-rs/rust-build/blob/main/idf-rust.Containerfile#L34-L52)

**Pull request merged**